src/script/conversation/ConversationRepository.test.ts

```diff
@@ -1262,5 +1265,340 @@ describe('ConversationRepository', () => {
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
+          {
+            members: {
+              others: [] as QualifiedId[],
+              self: {},
+            },
+            name: 'conv2',
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'staging.zinfra.io',
+              id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+            },
+            receipt_mode: 1,
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
+          {
+            members: {
+              others: [] as QualifiedId[],
+              self: {},
+            },
+            message_timer: null,
+            name: null,
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'anta.wire.link',
+              id: 'feabf90e-c785-577b-800a-556d8018542c',
+            },
+            receipt_mode: null,
+            team: null,
+            type: 2,
+          },
+        ],
+        failed: [
+          {
+            domain: 'staging.zinfra.io',
+            id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+          },
+          {
+            domain: 'staging.zinfra.io',
+            id: '05d0f240-bfe9-40d7-5678-602dac89fa1b',
+          },
+        ],
+      } as unknown as RemoteConversations;
+      const localConversations = [
+        {
+          id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
+          members: {
+            others: [] as QualifiedId[],
+            self: {},
+          },
+          name: 'conv1',
+          protocol: 'proteus',
+          qualified_id: {
+            domain: 'staging.zinfra.io',
+            id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
+          },
+          receipt_mode: 1,
+          team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+          type: 0,
+        },
+        {
+          id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+          members: {
+            others: [] as QualifiedId[],
+            self: {},
+          },
+          name: 'conv2',
+          protocol: 'proteus',
+          qualified_id: {
+            domain: 'staging.zinfra.io',
+            id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+          },
+          receipt_mode: 1,
+          team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+          type: 0,
+        },
+        {
+          id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+          members: {
+            others: [] as QualifiedId[],
+            self: {},
+          },
+          name: 'conv3',
+          protocol: 'proteus',
+          qualified_id: {
+            domain: 'staging.zinfra.io',
+            id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+          },
+          receipt_mode: 1,
+          team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+          type: 0,
+        },
+      ] as unknown as ConversationDatabaseData[];
+
+      const mergedConversation = ConversationMapper.mergeConversation(localConversations, remoteConversations);
```

Comment:
This test should go to the `ConversationMapper.test.ts` file, it doesn't belong in the `ConversationRepository` (since you never call the conversation repo in this test)